types/yup/index.d.ts

```diff
@@ -100,15 +96,20 @@ export interface MixedSchemaConstructor {
     new (options?: { type?: string; [key: string]: any }): MixedSchema;
 }
 
-// tslint:disable-next-line:no-empty-interface
-export interface MixedSchema extends Schema<any> {}
+export interface MixedSchema<T = any> extends Schema<T> {
+    nullable(isNullable?: true): MixedSchema<T | null>;
+    nullable(isNullable: false): MixedSchema<Exclude<T, null>>;
+    nullable(isNullable?: boolean): MixedSchema<T>;
```

Comment:
Instead of overloads, what about

```ts
nullable(isNullable?: boolean): MixedSchema<T extends true ? T | null : Exclude<T, null>>;

Reply:
@DanielRosenwasser 	

Sweet, I like removing the overloads, makes the definition more readable.

However when I try this works fine but using a wrapper function becomes harder. The [wrapper test code here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/36155/files#diff-3ed4607a6b19a001ccf1d37f42e3f387R663) needs to be updated to the following. 

```TypeScrript
function wrapper<T>(b: false, msx: MixedSchema<T>): MixedSchema<T>;
function wrapper<T>(b: true, msx: MixedSchema<T>): MixedSchema<T | null>;

function wrapper<T>(b: boolean, msx: MixedSchema<T>): MixedSchema<T extends true ? T | null : Exclude<T, null>> {
    return msx.nullable(b);
}

const resultingSchema1 = wrapper<string | number>(false, yup.mixed().oneOf(['1', 2])); // $ExpectType MixedSchema<string | number>
const resultingSchema2 = wrapper<string | number>(true, yup.mixed().oneOf(['1', 2])); // $ExpectType MixedSchema<string | number | null>
```

So the overloads are still required else the resulting type changes and the return of the wrapper function is a lot more verbose.

Personally I would favor more complexity in the type definition with a simple API over a simple type definition and a more complex API for developers to use. 

Lets put this to the vote and see what the others think and go with that.

Reply:
I agree with 👇 statement pretty wholeheartedly

> Personally I would favor more complexity in the type definition with a simple API over a simple type definition and a more complex API for developers to use.

I do wonder though, how many of our clients are using wrapper code like that because if that's somewhat of an "advanced" usecase, then I'm okay having the types be conditional so that we can get some of the benefits of simpler types on the type author side as well.